pkg/didcomm/protocol/middleware/presentproof/middlewares.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package presentproof
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/presentproof"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+	storeverifiable "github.com/hyperledger/aries-framework-go/pkg/store/verifiable"
+)
+
+const stateNamePresentationReceived = "presentation-received"
+
+// MetaData is an alias to the original MetaData
+type MetaData presentproof.MetaData
+
+// Provider contains dependencies for the SavePresentation middleware function
+type Provider interface {
+	VerifiableStore() storeverifiable.Store
+	VDRIRegistry() vdri.Registry
+}
+
+// SavePresentation the helper function for the present proof protocol which saves the presentations
+func SavePresentation(p Provider) presentproof.Middleware {
```

Comment:
this code looks a lot like `issuecredential.SaveCredentials()`. Did you think about extracting a common function?

Reply:
From my point of view, it will be more confused. It is not something reusable. I think that the final implementation can be a bit different. Probably the function will depend much more on the protocol.